src/lib/annotations/Annotator.js

```diff
@@ -241,93 +240,112 @@ class Annotator extends EventEmitter {
 
-        if (this.isInDrawMode()) {
-            this.toggleDrawAnnotationHandler();
+        // No specific mode available for annotation type
+        if (!(mode in this.modeButtons)) {
+            return;
         }
 
-        // If in annotation mode, turn it off
-        if (this.isInPointMode()) {
-            this.emit('annotationmodeexit');
-            this.annotatedElement.classList.remove(CLASS_ANNOTATION_POINT_MODE);
-            if (buttonEl) {
-                buttonEl.classList.remove(CLASS_ACTIVE);
-            }
+        const buttonSelector = this.modeButtons[mode].selector;
+        const buttonEl = event.target || this.previewUI.getAnnotateButton(buttonSelector);
 
-            this.unbindModeListeners(); // Disable point mode
-            this.bindDOMListeners(); // Re-enable other annotations
+        // Exit any other annotation mode
+        this.exitAnnotationModes(mode);
 
-            // Otherwise, enable annotation mode
+        // If in annotation mode, turn it off
+        if (this.isInAnnotationMode(mode)) {
+            this.disableAnnotationMode(mode, buttonEl);
+
+            // Remove annotation mode
+            this.currentAnnotatonMode = null;
         } else {
-            this.emit('annotationmodeenter', TYPES.point);
-            this.annotatedElement.classList.add(CLASS_ANNOTATION_POINT_MODE);
-            if (buttonEl) {
-                buttonEl.classList.add(CLASS_ACTIVE);
-            }
+            this.enableAnnotationMode(mode, buttonEl);
 
-            this.unbindDOMListeners(); // Disable other annotations
-            this.bindPointModeListeners(); // Enable point mode
+            // Update annotation mode
+            this.currentAnnotatonMode = mode;
         }
     }
 
     /**
-     * Toggles draw annotation mode on and off. When draw annotation mode is
-     * on, a click and draw
+     * Disables the specified annotation mode
      *
-     * @param {HTMLEvent} event - DOM event
+     * @param {string} mode - Current annotation mode
+     * @param {HTMLElement} buttonEl - Annotation button element
      * @return {void}
      */
-    toggleDrawAnnotationHandler(event = {}) {
-        this.destroyPendingThreads();
-        if (this.isInPointMode()) {
-            this.togglePointAnnotationHandler();
-        }
+    disableAnnotationMode(mode, buttonEl) {
+        this.emit('annotationmodeexit');
+        this.annotatedElement.classList.remove(CLASS_ANNOTATION_MODE);
+        if (buttonEl) {
+            buttonEl.classList.remove(CLASS_ACTIVE);
 
-        const drawButtonSelector = this.modeButtons[TYPES.draw].selector;
-        const buttonEl = event.target || this.previewUI.getAnnotateButton(drawButtonSelector);
-        const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
+            const drawEnterEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_ENTER);
+            annotatorUtil.hideElement(drawEnterEl);
```

Comment:
It is actually the case that the enterButton and drawButton are just SVGs under the same button. On mode change, we simply toggle the graphics.

On disable, we want to show the drawEnterEl (this button lets us enter draw mode), hide the cancelEl (exit draw mode without saving) and hide the postButtonEl (post is the hard commit button).